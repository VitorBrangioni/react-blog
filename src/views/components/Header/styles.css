header {
  grid-column-start: 1;
  grid-column-end: 3;
}

header {
  background-color: #212121;
  background-image: url(https://cdn2.hubspot.net/hubfs/3957645/blog%20image/Header_bg.png);
  background-color: #1e1e1e;
  background-size: cover;
  background-repeat: no-repeat;
}

header .container {
  align-items: center;
}

header img {
  width: 150px;
  margin-left: 20px;
  padding: 15px 0;
}

@media(max-width: 600px) {

  .one,
  .two,
  .three {
    background-color: #ffff;
    height: 5px;
    width: 100%;
    margin: 6px auto;
    transition-duration: 0.35s;
  }

  .menu-toggle {
    width: 40px;
    height: 60px;
    position: absolute;
    top: 25px;
    right: 30px;
  }

  .on {
    position: absolute;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background: #7159c1;
    z-index: 10;
  }

  .on .menu-toggle {
    position: absolute;
    right: 30px;
    top: 30px;  
  }

  .on .menu-toggle .one {
    transform: rotate(45deg) translate(7px, 7px);
  }
  .on .menu-toggle .two {
    opacity: 0;
  }
  .on .menu-toggle .three {
    transform: rotate(-45deg) translate(8px, -9px);
  }
}
