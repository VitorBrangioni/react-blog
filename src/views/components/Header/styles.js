import styled from "styled-components";

export const Header = styled.header`
  grid-column: 1/3;
  background-color: #212121;
  background-image: url(https://cdn2.hubspot.net/hubfs/3957645/blog%20image/Header_bg.png);
  background-color: #1e1e1e;
  background-size: cover;
  background-repeat: no-repeat;
`;

export const Container = styled.div`
  align-items: center;
`;
