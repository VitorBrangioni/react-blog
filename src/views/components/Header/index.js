import React, { Component } from "react";
// import './scriptMenu';
import "./styles.css";
import { Header } from "./styles";

class HeaderComponent extends Component {

  constructor() {
    super();
    this.state = {
      show: false
    }
  }

  render() {
    const { show } = this.state;

    return (
      <Header>
        <div className={`menu-section container ${!show || 'on' }`}>
          <img
            src="https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=454&name=Ativo%204.png%20454w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=907&name=Ativo%204.png%20907w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1361&name=Ativo%204.png%201361w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1814&name=Ativo%204.png%201814w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2268&name=Ativo%204.png%202268w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2721&name=Ativo%204.png%202721w"
            alt="test"
          />
          <div className="menu-toggle" onClick={this.handleToggleMenu.bind(this)}>
            <div className="one"></div>
            <div className="two"></div>
            <div className="three"></div>
          </div>
        </div>
      </Header>
    );
  }

  handleToggleMenu() {
    const { show } = this.state;
    this.setState({ show: !show });
    document.body.style.overflow = !show ? 'hidden' : 'initial';
  }
}

export default HeaderComponent;
