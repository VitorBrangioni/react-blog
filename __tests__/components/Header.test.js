import React from "react";
import Header from "../../src/views/components/Header";
import { render } from "@testing-library/react";

describe("Header", () => {
  it("Snapshot test", () => {
    const wrapper = render(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
