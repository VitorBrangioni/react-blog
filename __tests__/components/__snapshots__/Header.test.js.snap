// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-AxjAm fMkjKD"
      >
        <div
          class="menu-section container true"
        >
          <img
            alt="test"
            src="https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=454&name=Ativo%204.png%20454w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=907&name=Ativo%204.png%20907w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1361&name=Ativo%204.png%201361w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1814&name=Ativo%204.png%201814w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2268&name=Ativo%204.png%202268w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2721&name=Ativo%204.png%202721w"
          />
          <div
            class="menu-toggle"
          >
            <div
              class="one"
            />
            <div
              class="two"
            />
            <div
              class="three"
            />
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-AxjAm fMkjKD"
    >
      <div
        class="menu-section container true"
      >
        <img
          alt="test"
          src="https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=454&name=Ativo%204.png%20454w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=907&name=Ativo%204.png%20907w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1361&name=Ativo%204.png%201361w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1814&name=Ativo%204.png%201814w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2268&name=Ativo%204.png%202268w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2721&name=Ativo%204.png%202721w"
        />
        <div
          class="menu-toggle"
        >
          <div
            class="one"
          />
          <div
            class="two"
          />
          <div
            class="three"
          />
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
